experimental/unitsconversion/src/provider.rs

```diff
@@ -32,34 +33,46 @@ pub const KEYS: &[DataKey] = &[UnitsConstantsV1Marker::KEY];
 )]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
 #[yoke(prove_covariance_manually)]
-pub struct UnitsConstantsV1<'data> {
-    // TODO(#3882): Use a more efficient representation for the values with numerators and denominators.
-    // Also, the constant types.
-    /// Maps from constant name (e.g. ft_to_m) to the value of the constant (e.g. 0.3048).
+pub struct UnitsInfoV1<'data> {
+    /// Maps from unit name (e.g. foot) to the index of the unit in the `unit_quantity` vector.
     #[cfg_attr(feature = "serde", serde(borrow))]
-    pub constants_map: ZeroMap<'data, str, ConstantValueULE>,
+    pub units_info: ZeroMap<'data, str, UnitsInfoIndex>,
```

Comment:
Suggestion: You could possibly use a ZeroTrie here instead of a ZeroMap.

Reply:
Okay,  I want in the follow up PR to remove the `UnitsInfoIndex` and add `UnitsInfo` that has `Option<Dimension>` and `Option<ConversionInfo` instead of their indices. 

So, I will add this not to the issue too.

#4313